### ISSUE TITLE
Disabled entities still publish ComponentChangedMessage

### ISSUE THREAD

**Issue by DagobertDev** [Doraku/DefaultEcs#160]
If a component is Set on a disabled entity, the ComponentChangedMessage is still emitted.

**Comment by Doraku** [comment]
As you can query disabled entities too (`World.GetDisabledEntities()...`), this is by design. Is it causing some kind of problem?

**Comment by DagobertDev** [comment]
I'm extensively using subscription for behaviors in my game. Some subscriptions check for other components, like this:
```csharp
World.SubscribeComponentAdded((in Entity entity, in Foo foo) =>
{
    if (entity.Has<Bar>())
    {
        // Do something
    }
});
```
This works fine, but makes it impossible to clone entities with the `ComponentCloner`, because Foo might be copied before Bar, even if Bar will be there once the cloning finished. My idea was to disable the target entity, clone the components to it, and then enable it and trigger the notifications manually.
Now that i think about it, i notice that this approach will not work anyway, because there is no `Entity.NotifyAdded`. This means that i will need to figure out a different soluction, which will likely work independent of this issue.
It was surprising that the disabled entity still trigger notifications when components are cloned into it, maybe its worth documenting that notifications still trigger on disabled entities? Otherwise feel free to close it, as it doesn't affect me anymore.

**Comment by Doraku** [comment]
Before I added `World.GetDisabledEntities()` it was behaving as you said, no notification but after that it didn't make sens anymore to stop them. Take also `AResourceManager`, even if your entities are not enabled, you may want all the resources to be ready which is why the `SubscribeComponentAdded/Changed/Removed` need to be fired regardless of the entity status.
I think you may be able to use [EntitySet.EntityAdded](https://github.com/Doraku/DefaultEcs/blob/master/documentation/api/EntitySet_EntityAdded.md) by creating your set with a `WhenAdded<Foo>().WhenAdded<Bar>()` query. That way the callback would only trigger once each time your entity has both component added (or enabled, in case this is something you need to differentiate).

**Comment by Doraku** [comment]
Yeah maybe this use case is a little hidden. I may add a way to create subscriptions directly from an entity query so you don't have to keep a set alive (which will store the entities for nothing if you don't empty it). I try to keep things simple but the more I add the more complex it becomes ^^" I should also update the documentation to reflect that `World.Subscribe...` and the futur `World.GetEntities()...AsCallback` (name pending) are not exactly the same.
